packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +240,135 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def query_dns_server(fqdn, zone, local_ip)
+      query_dns = Resolv::DNS.new(nameserver: [local_ip], search: [zone], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      existing_ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      existing_serial = soa_record.serial if soa_record
+      new_serial = (existing_serial + 1).to_s if existing_serial
+      {
```

Comment:
I don't think this method should care about the 'new serial' and can be generated by whoever uses the output of this method.

Maybe the response should be:

```
{
 ip: a_record&.address,
 serial: soa_record&.serial
}
```

With the safe navigation operator we can safely call the instance methods even if the values of those objects is nil - but they do need to be defined. I think it's better having fewer if statements - therefore fewer execution paths thru the code. Omitted `existing` prefix be cause we can already assume that

Reply:
@jturel The `sat-deploy` box I tested on only had Ruby 2.0, which means the safe navigation operator isn't available :cry:

Reply:
Dang! My feedback on the output response is still relevant though Ditto for parse_zone_file